src/server/game/Maps/Map.cpp

```diff
@@ -3408,6 +3408,10 @@ bool Map::SpawnGroupSpawn(uint32 groupId, bool ignoreRespawn, bool force, std::v
 
     for (SpawnData const* data : toSpawn)
     {
+        // don't spawn if the current map difficulty is not used by the spawn
+        if (!(data->spawnMask & (1 << GetSpawnMode())))
```

Comment:
would it fit better in the for loop above, skipping calling `toSpawn.push_back(ASSERT_NOTNULL(data->ToSpawnData()));` ?

Reply:
It would yes, but I wanted to maintain code consistency and place this check to the other ones like the grid load check below here.